homeassistant/components/flume/coordinator.py

```diff
@@ -33,3 +41,53 @@ async def _async_update_data(self) -> None:
             self.flume_device.values,
             self.flume_device.query_payload,
         )
+
+
+class FlumeNotificationDataUpdateCoordinator(DataUpdateCoordinator[None]):
+    """Data update coordinator for flume notifications."""
+
+    def __init__(self, hass: HomeAssistant, auth) -> None:
+        """Initialize the Coordinator."""
+        super().__init__(
+            hass,
+            name=DOMAIN,
+            logger=_LOGGER,
+            update_interval=NOTIFICATION_SCAN_INTERVAL,
+        )
+        self.auth = auth
+        self.notifications: dict = {}
+        self.active_notifications_by_device: dict = {}
+
+    def _update_lists(self):
+        """Query flume for notification list."""
+        self.notifications = pyflume.FlumeNotificationList(
+            self.auth, read="true"
+        ).notification_list
+        _LOGGER.debug("Notifications %s", self.notifications)
+
+        notifications_by_device = {}
+
+        # Reformat notificatinos to correct format
+        for notification in self.notifications:
+            device_id = notification["device_id"]
+            extra = notification["extra"]
+            rule = notification["extra"]["event_rule_name"]
+
+            device_notifications = notifications_by_device.setdefault(device_id, {})
+            if rule == NOTIFICATION_BRIDGE_DISCONNECT:
+                # Bridge notifications are a special case
+                # both connect and disconnect register as notifications
+                # the last one (by time) will indicate connection state
+                device_notifications[rule] = extra["connected"]
+            else:
+                device_notifications[rule] = True
```

Comment:
Since you are mutating them here, maybe make `notifications_by_device` a `set()` instead and only add the ones that are in the notifying/non-default state.